### PR TITLE
Add `disableContentType` option, add an option to get upload progress, and add TypeScript support

### LINKED ISSUE #19: Unable to send multipart/form data with POST requests.
In [xhr level 2](https://dvcs.w3.org/hg/xhr/raw-file/tip/Overview.html), now widely supported , the [`XMLHttpRequest.send`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/send) method can be invoked with a [formdata](https://developer.mozilla.org/en-US/docs/Web/API/FormData) object, this will set the `Content-Type` header automatically to `multipart/form-data; boundary=----WebKitFormBoundaryxxxxxxxxxxx`.

Currently this library defaults the content type to `application/x-www-form-urlencoded` see [net/xhr.js l104](https://github.com/eBay/jsonpipe/blob/e75e3b58ef63f70deb6a91d46465e28adab3f5f5/lib/net/xhr.js#L104). Setting this causes xhr to not override the `content-type` header correctly and means that `multipart/form-data` cannot be sent with `POST` requests using this library.

Removing the default content-type fixes this.

### PR DESCRIPTION
**Pull request by foisonocean**
This PR contains several changes:

- Add `disableContentType` option to fix #19 
- Add an option to get upload progress
- Add TypeScript support

### CODE REVIEW

File: lib/net/xhr.js [hunk 1]
```diff
@@ -45,9 +45,16 @@ function send(url, options) {
         onHeaders = options.onHeaders || noop,
         errorFn = options.error || noop,
         completeFn = options.complete || noop,
+        disableContentType = options.disableContentType || false,
+        onUploadProgress = options.onUploadProgress,
         addContentHeader = method === 'POST',
         timer;
 
+    // Not all browsers support upload events
+    if (onUploadProgress != null && xhr.upload) {
```
Comment (senthilp) [inline lib/net/xhr.js]: Can we do a `typeof` check to see if it is a function, instead of just checking `null`

**Comment by senthilp** [comment]
Can you please update the documentation `README.md` to reflect the two new options.

**Comment by foisonocean** [comment]
@senthilp Updated. Please check.

**Comment by senthilp** [comment]
Thanks for the PR. Will publish a new version to npm.

**Comment by senthilp** [comment]
Published `jsonpipe@2.2.0`

**Pull request merged**